x-pack/plugins/reporting/server/usage/get_reporting_usage.ts

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { get } from 'lodash';
+import {
+  AggregationBuckets,
+  AggregationResults,
+  FeatureAvailabilityMap,
+  JobTypes,
+  KeyCountBucket,
+  RangeAggregationResults,
+  RangeStats,
+  UsageObject,
+} from './';
+import { decorateRangeStats } from './decorate_range_stats';
+// @ts-ignore untyped module
+import { getExportTypesHandler } from './get_export_type_handler';
+
+const JOB_TYPES_KEY = 'jobTypes';
+const JOB_TYPES_FIELD = 'jobtype';
+const LAYOUT_TYPES_KEY = 'layoutTypes';
+const LAYOUT_TYPES_FIELD = 'meta.layout.keyword';
+const OBJECT_TYPES_KEY = 'objectTypes';
+const OBJECT_TYPES_FIELD = 'meta.objectType.keyword';
+const STATUS_TYPES_KEY = 'statusTypes';
+const STATUS_TYPES_FIELD = 'status';
+
+const DEFAULT_TERMS_SIZE = 10;
+const PRINTABLE_PDF_JOBTYPE = 'printable_pdf';
+
+// indexes some key/count buckets by the "key" property
+const getKeyCount = (buckets: KeyCountBucket[]): { [key: string]: number } =>
+  buckets.reduce((accum, { key, doc_count: count }) => ({ ...accum, [key]: count }), {});
+
+function getAggStats(aggs: AggregationResults) {
+  const { buckets: jobBuckets } = aggs[JOB_TYPES_KEY] as AggregationBuckets;
+  const jobTypes = jobBuckets.reduce((accum, { key, doc_count: count }) => {
+    return {
+      ...accum,
+      [key]: {
+        total: count,
+      },
+    };
+  }, {}) as JobTypes;
+
+  // merge pdf stats into pdf jobtype key
+  const pdfJobs = jobTypes[PRINTABLE_PDF_JOBTYPE];
+  if (pdfJobs) {
+    const pdfAppBuckets = get(aggs[OBJECT_TYPES_KEY], '.pdf.buckets', []);
+    const pdfLayoutBuckets = get(aggs[LAYOUT_TYPES_KEY], '.pdf.buckets', []);
+    pdfJobs.app = getKeyCount(pdfAppBuckets) as {
+      visualization: number;
+      dashboard: number;
+    };
+    pdfJobs.layout = getKeyCount(pdfLayoutBuckets) as {
+      print: number;
+      preserve_layout: number;
+    };
+  }
+
+  const all = aggs.doc_count as number;
+  let statusTypes = {};
+  const statusBuckets = get(aggs[STATUS_TYPES_KEY], 'buckets', []);
+  if (statusBuckets) {
+    statusTypes = getKeyCount(statusBuckets) as {
+      completed: number;
+      failed: number;
```

Comment:
should pending  be included here?

Reply:
oops, obsolete hint for the compiler -- f99684b4b1